include/seqan/index/find2_index_approx.h

```diff
@@ -438,7 +553,8 @@ inline void _optimalSearchSchemeExact(TDelegate & delegate,
                                       TDir const & /**/,
                                       TDistanceTag const & /**/)
 {
-    bool goToRight2 = (blockIndex < s.pi.size() - 1) && s.pi[blockIndex + 1] > s.pi[blockIndex];
+    bool goToRight2 = (blockIndex < s.pi.size() - 1) ? s.pi[blockIndex + 1] > s.pi[blockIndex] :
```

Comment:
why did you replace this?

Reply:
It is not really necessary, since does not have an impact on the current version of the code, but if we are in the last block (of the search) and going to right, goToRight2 value was false.